src/vm/object.h

```diff
@@ -1205,6 +1209,32 @@ class ReflectClassBaseObject : public BaseObjectWithCachedData
 
 };
 
+class Utf8StringObject : public Object
+{
+#ifdef DACCESS_COMPILE
+    friend class ClrDataAccess;
+#endif
+
+private:
+    DWORD   m_StringLength;
+    BYTE    m_Characters[0];
+    // GC will see a Utf8StringObject like this:
+    //   DWORD m_StringLength
+    //   BYTE  m_Characters[0]
+    //   DWORD m_OptionalPadding (this is an optional field and will appear based on need)
```

Comment:
This comment about `m_OptionalPadding` does not make sense to me. Looks like left over from times when System.String was done differently.